pkg/ccl/backupccl/generative_split_and_scatter_processor.go

```diff
@@ -367,12 +370,28 @@ func runGenerativeSplitAndScatter(
 					return err
 				}
 				if chunkDestination == 0 {
-					// If scatter failed to find a node for range ingestion, route the range
-					// to the node currently running the split and scatter processor.
+					// If scatter failed to find a node for range ingestion, route the
+					// range to a random node that has already been scattered to so far.
+					// The random node is chosen by hashing the scatter key.
 					if nodeID, ok := flowCtx.NodeID.OptionalNodeID(); ok {
+						cachedNodeIDs := cache.cachedNodeIDs()
+						if len(cachedNodeIDs) > 0 && len(importSpanChunk.entries) > 0 {
+							hash := fnv.New32a()
+							if _, err := hash.Write(scatterKey); err != nil {
+								log.Warningf(ctx, "scatter returned node 0. Route span starting at %s to current node %v because of hash error: %v",
+									scatterKey, nodeID, err)
+							} else {
+								randomNum := int(hash.Sum32())
+								nodeID = cachedNodeIDs[randomNum%len(cachedNodeIDs)]
+							}
+
+							log.Warningf(ctx, "scatter returned node 0. "+
+								"Random route span starting at %s node %v", scatterKey, nodeID)
+						} else {
+							log.Warningf(ctx, "scatter returned node 0. "+
+								"Route span starting at %s to current node %v", scatterKey, nodeID)
```

Comment:
nit: could  you leave a todo along the lines of: OptionalNodeID only returns a node if the sql server runs in the same process as the kv server (e.g., not serverless). FIgure out how to properly handle this error in serverless restore.

Reply:
done